### PR TITLE
Use yarn and esbuild-register

### PR DESCRIPTION
**Pull request by ota-meshi**

### CODE REVIEW

**Review by JounQin — APPROVED** [review]

**Comment by JounQin** [comment]
Out of scope, do you think a constant `package.json` would help consistency across projects?

https://github.com/un-ts/prettier/tree/master/packages/pkg#sorting

File: .github/workflows/NodeCI.yml [hunk 1]
```diff
@@ -30,9 +28,9 @@ jobs:
       with:
         node-version: ${{ matrix.node-version }}
     - name: Install Packages
-      run: npm install --legacy-peer-deps
+      run: yarn install --ignore-engines
```
Comment (JounQin) [inline .github/workflows/NodeCI.yml:31]: @ota-meshi 

So in theory, `eslint-plugin-svelte@v2` does support node v12 well, is there any reason to enforce deprecating right now?

File: .github/workflows/NodeCI.yml [hunk 2]
```diff
@@ -46,12 +44,12 @@ jobs:
         node-version: ${{ matrix.node-version }}
     - name: Install @typescript-eslint v4
       run: |+
-        npm i -D @typescript-eslint/parser@4 @typescript-eslint/eslint-plugin@4 --legacy-peer-deps
-        npx rimraf node_modules
+        yarn add -D @typescript-eslint/parser@4 @typescript-eslint/eslint-plugin@4 --ignore-engines
+        rm -rf node_modules
```
Comment (JounQin) [inline .github/workflows/NodeCI.yml:48]: Is removing `node_modules` required? I don't see its benefit, actually.

Comment (ota-meshi) [reply .github/workflows/NodeCI.yml:48]: Previously It was sometimes confusing with `npm install` and couldn't be installed successfully. Therefore, it tries to keep it clean by deleting node_modules once.
We replaced it with yarn so we may not need it anymore.

Comment (JounQin) [reply .github/workflows/NodeCI.yml:48]: We can give it a try, deleting `node_modules` and reinstall may increase the CI running time.

**Comment by ota-meshi** [comment]
> Out of scope, do you think a constant package.json sorting would help consistency across projects?

I think a constant package.json sorting makes sense 👍

**Pull request merged**